helper/resource/testing_new_import_state.go

```diff
@@ -410,16 +434,45 @@ func appendImportBlock(config string, resourceName string, importID string) stri
 		resourceName, importID)
 }
 
+func appendImportBlockWithIdentity(config string, resourceName string, identityValues map[string]any) string {
+	configBuilder := config
+	configBuilder += fmt.Sprintf(``+"\n"+
+		`import {`+"\n"+
+		`	to = %s`+"\n"+
+		`	identity = {`+"\n",
+		resourceName)
+
+	for k, v := range identityValues {
+		configBuilder += fmt.Sprintf(`		%q = %q`+"\n", k, v)
+	}
+
+	configBuilder += `` +
+		`	}` + "\n" +
+		`}` + "\n"
+
+	return configBuilder
+}
+
 func importStatePreconditions(t testing.T, helper *plugintest.Helper, step TestStep) error {
 	t.Helper()
 
 	kind := step.ImportStateKind
 	versionUnderTest := *helper.TerraformVersion()
+	resourceIdentityMinimumVersion := version.Must(version.NewVersion("1.12.0-beta1"))
 
 	// Instead of calling [t.Fatal], we return an error. This package's unit tests can use [TestStep.ExpectError] to match
 	// on the error message. An alternative, [plugintest.TestExpectTFatal], does not have access to logged error messages,
 	// so it is open to false positives on this complex code path.
+	//
+	// Multiple cases may match, so check the most specific cases first
 	switch {
+	case kind.resourceIdentity() && versionUnderTest.LessThan(resourceIdentityMinimumVersion):
```

Comment:
In the interest of this being out-of-date sometime next week when they release `v1.12.0-beta2` 😆, should we just compare the core version directly here? Then we can get out of the game of verifying that they are using a specific pre-release version:

We have some precedence for doing this, specifically for the `tfversion` checks: https://github.com/hashicorp/terraform-plugin-testing/blob/bfd5be787814353a166ee6c5723d327b4839ccc3/tfversion/skip_below.go#L42-L53

Reply:
👍 I was looking for a simple way to compare the core versions. I'll try to borrow this.

Reply:
https://github.com/hashicorp/terraform-plugin-testing/pull/480/commits/9c6616f743b2715f3bd28abc284c860f2f5cc9c1